File: waste-reco-api/app/models/prediccion.py
from sqlalchemy import Column, Integer, String, ForeignKey, DateTime, Numeric
from sqlalchemy.orm import relationship
from models.base import Base

class Prediccion(Base):
    __tablename__ = "predicciones"

    id = Column(Integer, primary_key=True, index=True)
    precision_usuario = Column(Numeric(5, 4), index=True)
    tasa_acierto = Column(Numeric(5, 4), index=True)
    fecha_prediccion = Column(DateTime, index=True)
    id_residuo = Column(Integer, ForeignKey("residuos.id"), index=True)
    id_usuario = Column(Integer, ForeignKey("usuarios.id"), index=True)
    
    usuario = relationship("Usuario", backref="predicciones")
    residuo = relationship("Residuo", backref="predicciones")

File: waste-reco-web/src/models/Prediccion.ts
import type { Residuo } from './Residuo'
import type { Usuario } from './Usuario'

export default interface Reporte {
  id: number
  precision_usuario: number
  tasa_acierto: number
  fecha_prediccion: string
  tacho_esperado: boolean
  id_residuo: number
  id_usuario: number
}

export interface ReporteResponse {
  id: number
  precision_usuario: number
  tasa_acierto: number
  fecha_prediccion: string
  tacho_esperado: boolean
  residuo: Residuo
  usuario: Usuario
}

export const defaultReporte: Reporte = {
  id: 0,
  precision_usuario: 0,
  tasa_acierto: 0,
  fecha_prediccion: '',
  tacho_esperado: false,
  id_residuo: 0,
  id_usuario: 0,
}
